### PR TITLE
Make TabixReader take a hadoop configuration in the constructor

### PR DESCRIPTION
**Pull request by chrisvittal**
This fixes a bug in import_vcfs as reading the indices and generating
partitions is parallelized.

### CODE REVIEW

**Comment by chrisvittal** [comment]
I discovered this when I tried to run a vcf combiner pipeline. To me, this signals that we need better knowledge of where integration tests live and how to add to them.

**Comment by tpoterba** [comment]
what is the bug?

**Comment by tpoterba** [comment]
Ah.

**Comment by tpoterba** [comment]
Can we have a test that catches it?

**Comment by tpoterba** [comment]
Hmm, maybe this would always work in local mode since they're in a shared JVM?

**Comment by tpoterba** [comment]
That's hard to test.

**Comment by chrisvittal** [comment]
Exactly, this always works in local mode. We definitely need more ways to test behavior in the non-local environment, but I have no concrete ideas right now.

**Comment by danking** [comment]
You can write non-local tests by [submitting a python file](https://github.com/hail-is/hail/blob/master/hail/hail-ci-build.sh#L244-L245) to the cluster that is started by the CI.

**Comment by tpoterba** [comment]
We should really be running all of our tests on a cluster. We can run Spark in cluster mode on a single machine, that's probably what we should do

**Comment by danking** [comment]
That still leaves open problems caused by true network communication between physically-distant cores. We could package our tests as a test JAR and submit that to the Spark cluster the CI starts.

**Comment by tpoterba** [comment]
What kinds of problems are in that category? I'm not saying they don't exist, but shouldn't it be difficult to cause that kind of problem given our base abstractions?

**Comment by tpoterba** [comment]
It's *not* easy to make `HailContext.get` work on the workers.

**Comment by danking** [comment]
@tpoterba The only things that come to mind are shuffle issues and shared filesystem bugs. Regardless we already start a cluster and it's not hard to use that existing functionality run a non-local test.

**Comment by tpoterba** [comment]
putting a test in hail-ci-build seems like the wrong thing. We should just run all our tests against a cluster-mode spark, no?

**Comment by chrisvittal** [comment]
I believe that even a local cluster (2+ jvms) would be sufficient to reproduce this error. I just have no idea how to configure such a thing.

**Comment by danking** [comment]
I also do not know how to run our tests in cluster-mode, but I know how to add a python file to this repo and submit it to the cluster in hail-ci-build.sh ;)

**Comment by chrisvittal** [comment]
Alright, let's see how this does.

File: hail/hail-ci-build.sh [hunk 1]
```diff
@@ -241,11 +241,9 @@ test_gcp() {
          --zip gs://hail-ci-0-1/temp/$SOURCE_SHA/$TARGET_SHA/hail.zip \
          --vep
 
-    time cluster submit ${CLUSTER_NAME} \
-         cluster-sanity-check.py
-
-    time cluster submit ${CLUSTER_NAME} \
-         cluster-vep-check.py
+    for script in python/cluster-tests/**.py; do
```
Comment (danking) [inline hail/hail-ci-build.sh]: v. nice.

File: hail/python/cluster-tests/cluster-read-vcfs-check.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import json
+import hail as hl
+
+gvcfs = ['gs://hail-ci/gvcfs/HG00096.g.vcf.gz',
```
Comment (danking) [inline hail/python/cluster-tests/cluster-read-vcfs-check.py]: did you manually upload these?

Comment (chrisvittal) [reply hail/python/cluster-tests/cluster-read-vcfs-check.py]: Yep, did it earlier.

**Comment by chrisvittal** [comment]
It works! That same script fails if you try to submit it to a cluster running current master.

**Pull request merged**